File: backend/src/main/java/finki/ukim/backend/administration/service/domain/StaffService.java
package finki.ukim.backend.administration.service.domain;

import finki.ukim.backend.administration.model.domain.Staff;
import finki.ukim.backend.administration.model.projection.StaffPageableProjection;
import finki.ukim.backend.auth_and_access.model.domain.User;
import finki.ukim.backend.auth_and_access.model.projection.UserPageableProjection;
import org.springframework.data.domain.Page;

import java.util.List;

public interface StaffService {
    List<Staff> findAll(User currentUser);

    List<UserPageableProjection> findUsersAvailableForStaff(User currentUser);

    List<Staff> findAllWithAll(User currentUser);

    Staff findById(Long id, User currentUser);

    Staff find(Long userId, Long departmentId, Long municipalityId);

    Staff findByUserId(User currentUser, Long userId);

    List<Staff> findByDepartmentId(Long departmentId);

    List<Staff> findByMunicipalityId(Long municipalityId);

    Staff create(User currentUser, Staff staff);

    Staff update(Long id, Staff updatedStaff);

    Staff deleteById(Long id, User currentUser);

    Page<StaffPageableProjection> findAll(
            User user,
            int page,
            int size,
            String sortBy,
            Long id,
            Long userId,
            Long departmentId,
            Long municipalityId,
            String username,
            String municipalityCode,
            String municipalityName
    );
}

File: frontend/src/administration/service/staffService.js
import axiosInstance from "../../common/axios/axios.js";

const staffService = {
    findAll: async () => {
        return await axiosInstance.get("/staff");
    },

    findAllPaged: async ({
                             page = 0,
                             size = 10,
                             sortBy = "id",
                             id = null,
                             userId = null,
                             departmentId = null,
                             municipalityId = null,
                             username = null,
                             municipalityCode = null,
                             municipalityName = null
                         }) => {
        return await axiosInstance.get("/staff/paged", {
            params: {
                page,
                size,
                sortBy,
                id,
                userId,
                departmentId,
                municipalityId,
                username,
                municipalityCode,
                municipalityName
            }
        });
    },

    findAllDetailed: async () => {
        return await axiosInstance.get("/staff/detailed");
    },

    findById: async (id) => {
        return await axiosInstance.get(`/staff/${id}`);
    },

    findByUserId: async (userId) => {
        return await axiosInstance.get(`/staff/by-user/${userId}`);
    },

    findByDepartmentId: async (departmentId) => {
        return await axiosInstance.get(`/staff/by-department/${departmentId}`);
    },

    findByMunicipalityId: async (municipalityId) => {
        return await axiosInstance.get(`/staff/by-municipality/${municipalityId}`);
    },

    findUsersAvailableForStaff: async () => {
        return await axiosInstance.get("/staff/available-users");
    },

    create: async (data) => {
        return await axiosInstance.post("/staff", data);
    },

    update: async (id, data) => {
        return await axiosInstance.put(`/staff/${id}`, data);
    },

    deleteById: async (id) => {
        return await axiosInstance.delete(`/staff/${id}`);
    }
};

export default staffService;